modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+
+. Check the instrumentation object is created:
++
+[source,console]
+----
+oc get instrumentation -n <your-project>
+----
+
+. Verify that the resources were deployed in the correct order
++
+Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.
+
+When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
+----
+
+You should see something like this:
+
+[source,console]
+----
+... Created container opentelemetry-auto-instrumentation
+... Started container opentelemetry-auto-instrumentation
+----
+
+If those messages are missing, then it means that was an issue while injecting the auto-instrumentation.
+
+. Inspect for the operator logs for instrumentation errors:
++
+[source,console]
+----
+oc logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n openshift-opentelemetry-operator --follow
+----
+
+. Confirm that the application pods you want to instrument are labeled correctly and the appropriate auto-instrumentation settings have been applied. For example, if you are instrumenting a Python application, you can check that the pods have the `instrumentation.opentelemetry.io/inject-python='true'` annotation:
++
+[source,console]
+----
+oc get pods -n <your-project> -o jsonpath='{range .items[?(@.metadata.annotations["instrumentation.opentelemetry.io/inject-python"]=="true")]}{.metadata.name}{"\n"}{end}'
+----
+
+You should double-check the annotation applied to the instrumentation object is the correct one for the language you are instrumenting. If there are multiple instrumentations in the same namespace, you should set its name in the annotation. Also, if the Instrumentation object is in a different namespace, you need to specify it in the annotation. You can find the accepted values in the link:otel-instrumentation.html["OpenTelemetry instrumentation configuration options"] section.
+
+When defining a Deployment, remember that annotations can be added in either {{spec.metadata.annotations}} or {{spec.template.metadata.annotations}}. Ensure that the auto-instrumentation annotation is added to {{spec.template.metadata.annotations}}.
```

Comment:
IOW, is it correct to say that:
1. Auto-instrumentation must be defined only in `spec.template.metadata.annotations`.
2. If auto-instrumentation was defined in `spec.metadata.annotations`, then that is a mistake (that causes some error), and this mistake must be corrected by defining auto-instrumentation in `spec.template.metadata.annotations`?

Reply:
1. The annotation, yes.
2. There will not be an error. Nothing will happen. Which is somehow an error because the user will want something to happen -where something means the instrumentation injected in the workload-.

Reply:
Done.